### PR TITLE
feat(docs): document `hex` package

### PR DESCRIPTION
**Pull request by pierwill**

### CODE REVIEW

File: stdlib/contrib/bonitoo-io/hex/hex.flux [hunk 1]
```diff
@@ -1,7 +1,31 @@
...
+// ## Parameters
+//
+// - v: string to convert.
 builtin string : (v: A) => string
+
+// uint converts a hexadecimal string to an unsigned integer.
+//
+// ## Parameters
+//
+// - v: string to convert.
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/hex/hex.flux:23]:
```suggestion
// - v: String to convert.
```

File: stdlib/contrib/bonitoo-io/hex/hex.flux [hunk 2]
```diff
@@ -1,7 +1,31 @@
...
 builtin uint : (v: string) => uint
+
+// bytes converts a hexadecimal string to bytes.
+//
+// ## Parameters
+//
+// - v: string to convert.
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/hex/hex.flux:30]:
```suggestion
// - v: String to convert.
```

File: stdlib/contrib/bonitoo-io/hex/hex.flux [hunk 3]
```diff
@@ -1,7 +1,31 @@
+// Package hex provides functions that perform hexadecimal conversion
+// of `int`, `uint` or `bytes` values to and from `string` values.
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/hex/hex.flux:2]:
```suggestion
// of `int`, `uint` or `bytes` values to and from `string` values.
//
// introduced: 0.131.0
```

File: stdlib/contrib/bonitoo-io/hex/hex.flux [hunk 4]
```diff
@@ -1,7 +1,31 @@
...
 
-
+// int converts a hexadecimal string to an integer.
+//
+// ## Parameters
+//
+// - v: string to convert.
 builtin int : (v: string) => int
+
+// string converts an integer string to a hexadecimal string.
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/hex/hex.flux:12]:
```suggestion
// string converts a Flux basic type to a hexadecimal string.
//
// The function is similar to `string()`, but encodes int, uint, and bytes
// types to hexadecimal lowercase characters.
```

**Review by sanderson — CHANGES_REQUESTED** [review]
We need to port all the **examples** and **tags** from the [hex package docs](https://docs.influxdata.com/flux/v0.x/stdlib/contrib/bonitoo-io/hex/).

**Comment by pierwill** [comment]
Bad git happening here...

**Comment by pierwill** [comment]
Still need to port examples.

File: stdlib/contrib/bonitoo-io/hex/hex.flux [hunk 5]
```diff
@@ -1,7 +1,77 @@
...
 
-
+// int converts a hexadecimal string to an integer.
+//
+// ## Parameters
+//
+// - v: String to convert.
+//
+// ## Examples
+//
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/hex/hex.flux:14]: Each example needs an h3 title.

File: stdlib/contrib/bonitoo-io/hex/hex.flux [hunk 6]
```diff
@@ -1,7 +1,77 @@
...
+// The function is similar to `string()`, but encodes int, uint, and bytes
+// types to hexadecimal lowercase characters.
+//
+// ## Parameters
+//
+// - v: Value to convert.
+//
+// ## Examples
+//
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/hex/hex.flux:35]: This function in particular has many examples: https://docs.influxdata.com/flux/v0.x/stdlib/contrib/bonitoo-io/hex/string/#examples

File: stdlib/contrib/bonitoo-io/hex/hex.flux [hunk 7]
```diff
@@ -1,7 +1,156 @@
...
+// import "contrib/bonitoo-io/hex"
+//
+// hex.int(v: "4d2")
+//
+// // Returns 1234
+// ```
 builtin int : (v: string) => int
+
+// string converts a Flux basic type to a hexadecimal string.
+// tag: type-conversion
```
Comment (nathanielc) [inline stdlib/contrib/bonitoo-io/hex/hex.flux:26]: these tags and other metadata must be the last lines of the comment or they will not be parsed.

File: stdlib/contrib/bonitoo-io/hex/hex.flux [hunk 8]
```diff
@@ -1,7 +1,156 @@
...
+// - v: String to convert.
+//
+// ## Examples
+// ### Convert string into bytes
+// ```
+// import "contrib/bonitoo-io/hex"
+//
+// hex.bytes(v: "FF5733")
+//
+// Returns [255 87 51] (bytes)
```
Comment (nathanielc) [inline stdlib/contrib/bonitoo-io/hex/hex.flux:154]: nit, missing extra `//`

**Review by nathanielc — CHANGES_REQUESTED** [review]

**Review by nathanielc — APPROVED** [review]

File: stdlib/contrib/bonitoo-io/hex/hex.flux [hunk 9]
```diff
@@ -1,7 +1,157 @@
...
+// 
+// ### Convert a boolean to a hexadecimal string value
+// ```
+// import "contrib/bonitoo-io/hex"
+// 
+// hex.string(v: true)
+// 
+// // Returns "true"
+// ```
+// ### Convert a duration to a hexadecimal string value
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/hex/hex.flux:53]:
~~~suggestion
// ```
//
// ### Convert a duration to a hexadecimal string value
~~~

File: stdlib/contrib/bonitoo-io/hex/hex.flux [hunk 10]
```diff
@@ -1,7 +1,157 @@
...
+// 
+// ### Convert a uinteger to a hexadecimal string value
+// ```
+// import "contrib/bonitoo-io/hex"
+// 
+// hex.string(v: uint(v: 5678))
+// 
+// // Returns "162e"
+// ```
+// ### Convert a float to a hexadecimal string value
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/hex/hex.flux:88]:
~~~suggestion
// ```
//
// ### Convert a float to a hexadecimal string value
~~~

File: stdlib/contrib/bonitoo-io/hex/hex.flux [hunk 11]
```diff
@@ -1,7 +1,157 @@
...
+// ```
+// import "contrib/bonitoo-io/hex"
+// 
+// hex.string(v: 10.12)
+// 
+// // Returns "10.12"
+// ```
+// ### Convert bytes to a hexadecimal string value
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/hex/hex.flux:96]:
~~~suggestion
// ```
//
// ### Convert bytes to a hexadecimal string value
~~~

File: stdlib/contrib/bonitoo-io/hex/hex.flux [hunk 12]
```diff
@@ -1,7 +1,157 @@
...
+// ```
+// import "contrib/bonitoo-io/hex"
+// 
+// hex.string(v: bytes(v: "Hello world!"))
+// ```
+// 
+// // Returns "48656c6c6f20776f726c6421"
+//
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/hex/hex.flux:104]:
~~~suggestion
// // Returns "48656c6c6f20776f726c6421"
// ```
// 
~~~

File: stdlib/contrib/bonitoo-io/hex/hex.flux [hunk 13]
```diff
@@ -1,7 +1,157 @@
...
+// 
+// hex.string(v: bytes(v: "Hello world!"))
+// ```
+// 
+// // Returns "48656c6c6f20776f726c6421"
+// 
+// ### Convert all values in a column to hexadecimal string values
+// Use map() to iterate over and update all input rows.
+// Use hex.string() to update the value of a column.
+// The following example uses data provided by the sampledata package.
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/hex/hex.flux:108]:
```suggestion
// Use `map()` to iterate over and update all input rows.
// Use `hex.string()` to update the value of a column.
// The following example uses data provided by the sampledata package.
```

File: stdlib/contrib/bonitoo-io/hex/hex.flux [hunk 14]
```diff
@@ -1,7 +1,157 @@
...
+// 
+// ```
+// import "sampledata"
+// import "contrib/bonitoo-io/hex"
+// 
+// data = sampledata.int()
+//   |> map(fn: (r) => ({ r with _value: r._value * 1000 }))
+// 
+// data
+//   |> map(fn:(r) => ({ r with _value: hex.string(v: r.foo) }))
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/hex/hex.flux:118]:
```suggestion
// < data
// >     |> map(fn:(r) => ({ r with _value: hex.string(v: r.foo) }))
```

File: stdlib/contrib/bonitoo-io/hex/hex.flux [hunk 15]
```diff
@@ -1,7 +1,157 @@
...
+// Use map() to iterate over and update all input rows.
+// Use hex.string() to update the value of a column.
+// The following example uses data provided by the sampledata package.
+// 
+// ```
+// import "sampledata"
+// import "contrib/bonitoo-io/hex"
+// 
+// data = sampledata.int()
+//   |> map(fn: (r) => ({ r with _value: r._value * 1000 }))
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/hex/hex.flux:115]:
```suggestion
//     |> map(fn: (r) => ({ r with _value: r._value * 1000 }))
```

File: stdlib/contrib/bonitoo-io/hex/hex.flux [hunk 16]
```diff
@@ -1,7 +1,157 @@
...
 builtin string : (v: A) => string
+
+// uint converts a hexadecimal string to an unsigned integer.
+//
+// ## Parameters
+//
+// - v: String to convert.
+//
+// ## Examples
+// ### Convert hexadecimal string to unsigned integer
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/hex/hex.flux:130]:
```suggestion
// ### Convert a hexadecimal string to an unsigned integer
```

File: stdlib/contrib/bonitoo-io/hex/hex.flux [hunk 17]
```diff
@@ -1,7 +1,157 @@
...
 builtin uint : (v: string) => uint
+
+// bytes converts a hexadecimal string to bytes.
+//
+// ## Parameters
+//
+// - v: String to convert.
+//
+// ## Examples
+// ### Convert string into bytes
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/hex/hex.flux:148]:
```suggestion
// ### Convert a hexadecimal string into bytes
```

**Review by sanderson — CHANGES_REQUESTED** [review]

**Review by sanderson — APPROVED** [review]

**Pull request merged**